### ISSUE TITLE
asyncio support

### ISSUE THREAD

**Issue by Gr1N** [sdispater/orator#113]
Hi!

Thanks for your work guys, ORM looks very nice to me (good replacement for `SQLAlchemy` to my mind). 

But my question about `asyncio` support (or your plans to support if any), I found nothing in docs, but maybe I wrong. It would be great to have possibility to write code like this (https://github.com/aio-libs/aiopg#example-of-sqlalchemy-optional-integration) using `orator`.

**Comment by sdispater** [comment]
Thanks for your interest in Orator!

Orator does not currently support asyncio but that's definitely something I want to do.

I don't know just yet if it will be directly integrated in the library or available as a separate package.

**Comment by Gr1N** [comment]
Nice to hear this! 

I know this is weird question, but maybe you have any estimates (like half year, year...)?

**Comment by MechanisM** [comment]
I see that aiopg mentioned, but there is better option [`asyncpg`](https://github.com/MagicStack/asyncpg)

**Comment by sdispater** [comment]
@Gr1N No, I don't have any estimates yet.

**Comment by jychen7** [comment]
Nice to hear this.
I think separate package would be better, like https://github.com/challengepost/activeasync

**Comment by JosXa** [comment]
Same here, just trying out the library coming from a Peewee background and wondering if there are any plans to make Orator asynchronous

**Comment by josephmancuso** [comment]
@mitchfriedman @JosXa Out of curiosity and as someone not familiar with exactly what parts would be asyncio, what are you thinking to make asynchronous? Is it just saving models asynchronously?

**Comment by mitchfriedman** [comment]
I'm not sure exactly what is needed to be changed to make this compatible with asyncio, but yeah anytime an IO operation (in this case, not just saving, but all calls to the DB - connection creation/resume/termination, all queries, etc) need to yield execution to the event loop.

**Comment by mrcartoonster** [comment]
I agree in making it a seperate package and to simplify the async, you can use [Trio](https://trio.readthedocs.io/en/stable/index.html) and it's postgres library [trioppg](https://github.com/python-trio/triopg).
